In [1]:
import numpy as np
import matplotlib.pyplot as plt

In [2]:
#N=100

m=2
N = m*50
d = 5
L = 5*d

T = np.zeros(5*N)
x = np.linspace(0,L,5*N)


In [3]:
def Tfunc(T):
    for i in range(len(T)):
        if x[i]>10 and x[i]<=15:
            T[i]= 1230+273 # 1150 in Kelvin
        else:
            T[i]=20+273 # 0 Deg in Kelvin
    return T

T = Tfunc(T)
#print(T)


In [4]:
k = 5.2 #J/m/s/K
c = 1300 #J/kg/K
p = 2750 #kg/m^3
delta_t = 1800//(m**2) #sec
delta_x = (L/(N*5))**2
n = (delta_t/delta_x)*(k/(c*p)) 

print("n:", n)

def tchange(T,n):
    Tnew=np.copy(T)
    for i in range(1,N*5-1):
        Tnew[i] = n*((T[i+1])-(2*T[i])+T[i-1])+(T[i])
        Tnew = T
    return T

#print(tchange(T,n))


n: 0.2618181818181818


In [5]:
#establish  times 

t_org = 1800

hour = 60*60 
day = 60*60*24
week = 60*60*24*7
month = 60*60*24*30
year = 60*60*24*365

time = np.array([60*60 , 60*60*24, 60*60*24*7, 60*60*24*30, 60*60*24*365])
timecalc = time//delta_t

for t in timecalc:
    T = Tfunc(T)
    for j in range(t):
        T=tchange(T,n)
    print(max(T))
    
    '''
    plt.plot(x,T,'ro-',mfc='w')
    plt.ylim(273,1500)
    plt.show()    
    '''
        



1503.0
1502.9761763417946
1375.5219658756034
977.7841270139874
470.78794744369816


In [6]:
#N=200

m=4
N = m*50
d = 5
L = 5*d

T = np.zeros(5*N)
x = np.linspace(0,L,5*N)

def Tfunc(T):
    for i in range(len(T)):
        if x[i]>10 and x[i]<=15:
            T[i]= 1230+273 # 1150 in Kelvin
        else:
            T[i]=20+273 # 0 Deg in Kelvin
    return T

T = Tfunc(T)
#print(T)

k = 5.2 #J/m/s/K
c = 1300 #J/kg/K
p = 2750 #kg/m^3
delta_t = 1800//(m**2) #sec
delta_x = (L/(N*5))**2
n = (delta_t/delta_x)*(k/(c*p)) 

print("n:", n)

def tchange(T,n):
    Tnew=np.copy(T)
    for i in range(1,N*5-1):
        Tnew[i] = n*((T[i+1])-(2*T[i])+T[i-1])+(T[i])
        Tnew = T
    return T

#print(tchange(T,n))

n: 0.2606545454545454


In [7]:
t_org = 1800

hour = 60*60 
day = 60*60*24
week = 60*60*24*7
month = 60*60*24*30
year = 60*60*24*365

time = np.array([60*60 , 60*60*24, 60*60*24*7, 60*60*24*30, 60*60*24*365])
timecalc = time//delta_t

for t in timecalc:
    T = Tfunc(T)
    for j in range(t):
        T=tchange(T,n)
    print(max(T))

1503.0
1502.9778681669927
1375.8342654730084
978.2087145556623
471.42857697165573


### Question 2

| cooling time | CONVERGED Temperature at the dike center (K) |
| --- | --- |
| 1 hour | 1503 |
| 1 day | 1503 |
| 1 week | 1375 |
| 1 month | 978 |
| 1 year |  471  |